pkg/policy/mapstate_test.go

```diff
@@ -1341,6 +1342,38 @@ func TestMapState_AccumulateMapChanges(t *testing.T) {
 			egressKey(43, 6, 80, 0): {},
 		},
 		deletes: Keys{},
+	}, {
+		continued: false,
+		name:      "test-5c - proxy port propagation from the most specific covering key with a proxy port",
+		args: []args{
+			{cs: csFoo, adds: []int{43}, port: 80, proto: 6, prefix: 12, redirect: true},
+			{cs: csBar, adds: []int{43}, port: 80, proto: 6, prefix: 16, authReq: AuthTypeSpire.AsExplicitRequirement()},
+		},
+		state: testMapState(mapStateMap{
+			egressKey(43, 6, 80, 12): proxyEntry(1),
+			egressKey(43, 6, 80, 0):  proxyEntry(1).withExplicitAuth(AuthTypeSpire),
+		}),
+		adds: Keys{
+			egressKey(43, 6, 80, 12): {},
+			egressKey(43, 6, 80, 0):  {},
+		},
+		deletes: Keys{},
+	}, {
+		continued: false,
+		name:      "test-5d - proxy port propagation from the most specific covering key with a proxy port - reverse",
+		args: []args{
+			{cs: csBar, adds: []int{43}, port: 80, proto: 6, prefix: 16, authReq: AuthTypeSpire.AsExplicitRequirement()},
+			{cs: csFoo, adds: []int{43}, port: 80, proto: 6, prefix: 12, redirect: true},
```

Comment:
want to add another entry with longer prefix, just to make sure it's carried down correctly?

Reply:
Yes, and found issues while doing so!